product_catalog_aeroo_report/report/product_catalog_parser.py

```diff
@@ -72,10 +73,34 @@ def get_price(self, product, pricelist, context=None):
             context = {}
         context['pricelist'] = pricelist.id
         product_obj = self.pool[self.product_type]
+        sale_uom = self.pool['product.template'].fields_get(
+            self.cr, self.uid, ['sale_uom_ids'])
+        if sale_uom and product.sale_uom_ids:
+                context['uom'] = product.sale_uom_ids[0].uom_id.id
         price = product_obj.browse(
             self.cr, self.uid, [product.id], context=context).price
         return price
 
+    def get_description(self, product, print_product_uom, context=None):
+
+        sale_uom = self.pool['product.template'].fields_get(
+            self.cr, self.uid, ['sale_uom_ids'])
+        name_uoms = ''
+        if not print_product_uom:
+            return product.name
+
+        if sale_uom and product.sale_uom_ids:
+            for x in product.sale_uom_ids:
+                name_uoms = name_uoms + x.uom_id.name
+            description = product.name + \
```

Comment:
Nico, toda esta parte hacela traducible (y hacete la traducción) y hacelta con los comodines, algo tipo esto (además uso mapped y join para unir el listado de unidades de medida, mucho mas facil) y además solo agregamos eso si hay más de una uom_ids
     if sale_uom and product.sale_uom_ids:
            main_uom = product.sale_uom_ids[0].uom_id
     else:
            main_uom = product.uom_id
     description = _('%s (%s)" % (product.name, main_uom.name)
     if len(product.sale_uom_ids) > 1:
            description = _("%s. Also available in %s') % (description, ', '.join(product.sale_uom_ids.mapped('uom_id.name')))